### DISCUSSION TITLE
WebServerControl for multiple sub apps

### DISCUSSION THREAD

**Discussion by ohagenson** [idaholab/moose#27624] [Q&A Tools] [answered]
Hello,

I recently found out about the new WebServerControl feature and the Python MooseControl implementation.

I wanted to possibly use this feature to couple a simulation of aquifer thermal energy storage in MOOSE with other simulations and the Functional Mock-Up Interface in Python.
For both the hot and the cold well, I wanted to set up a separate app and dynamically switch on/off BCs in Python depeding on the wished operation.
My question here is whether it is possible to control multiple apps from Python? Or do I have to set up a parent multi-app that controls both apps and control it via Python's MooseControl?

And another small question: Is it possible to access the time step and the time step width via MooseControl in order to avoid the loop over a fixed number of steps in Python?

Thanks!

**Comment by GiudGiud** [comment, marked as answer]
Hello

> My question here is whether it is possible to control multiple apps from Python? Or do I have to set up a parent multi-app that controls both apps and control it via Python's MooseControl?

the latter is what you need to do.
Send a signal to control the activation of the BC from the main app, and modify that signal from the MooseControl.
I would use postrpcessors.
A ConstantPostprocessor on the main app, its value is controllable
A Receiver on the child app, to which the 'value' from the ConstantPP is being transferred using a MultiAppPostprocessorTransfer
